### PR TITLE
[metal] Add int4mm weight packing mps kernel, and improved int4mm shader

### PR DESCRIPTION
**Pull request by manuelcandales**
Adds _convert_weight_to_int4pack MPS kernel
Replaces previous int4mm Metal shader, with shader authored by @kimishpatel  which improves perf by ~40%

### CODE REVIEW

File: test/test_mps.py [hunk 1]
```diff
@@ -1617,14 +1617,6 @@ def test_exp(self, device="mps", dtype=torch.float):
             a = torch.tensor(v, dtype=dtype, device="mps") * b
             self.compare_with_numpy(torch.exp, np.exp, a)
 
-    def test_triu_inf(self, device="mps", dtype=torch.float):
-        for diag in [-1, 0, 1]:
-            mask = torch.full((3, 6, 6), float("-inf"))
-            mask_mps = mask.clone().detach().to('mps')
-            cpu_ref = torch.triu(mask, diagonal=diag)
-            mps_out = torch.triu(mask_mps, diagonal=diag)
-            self.assertEqual(cpu_ref, mps_out)
```
Comment (malfet) [inline test/test_mps.py:1626]: Why are you deleting this thing?

**Review by malfet — APPROVED** [review]
LGTM, but please do not delete the test

File: test/test_mps.py (hunk 1 above)
Comment (manuelcandales) [reply test/test_mps.py:1626]: Oops, that was unintended. Fixing that.

**Pull request closed without merging**